ts/components/ChooserListContainer.tsx

```diff
@@ -0,0 +1,253 @@
+import color from "color";
+import { none, Option, some } from "fp-ts/lib/Option";
+import I18n from "i18n-js";
+import { debounce } from "lodash";
+import { Body, Button, Content, Input, Item, Right, View } from "native-base";
+import * as React from "react";
+import {
+  ImageSourcePropType,
+  KeyboardAvoidingView,
+  ListRenderItem,
+  Platform,
+  StyleSheet
+} from "react-native";
+import customVariables from "../theme/variables";
+import variables from "../theme/variables";
+import ChooserList from "./ChooserList";
+import ChooserListSearch from "./ChooserListSearch";
+import AppHeader from "./ui/AppHeader";
+import FooterWithButtons from "./ui/FooterWithButtons";
+import IconFont from "./ui/IconFont";
+
+type Props<T> = {
+  items: ReadonlyArray<T>;
+  keyExtractor: (item: T, index: number) => string;
+  renderItem: ListRenderItem<T>;
+  onCancel: () => void;
+  isRefreshEnabled: boolean;
+  isRefreshing?: boolean;
+  onRefresh?: () => void;
+  isSearchEnabled: boolean;
+  onSearchItemContainsText?: (item: T, searchText: string) => boolean;
+  noSearchResultsSourceIcon?: ImageSourcePropType;
+  noSearchResultsSubtitle?: string;
+};
+
+type State = {
+  searchText: Option<string>;
+  debouncedSearchText: Option<string>;
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  },
+  content: {
+    flex: 1
+  },
+  headerNoSearch: {
+    height: customVariables.contentPadding / 2
+  }
+});
+
+/**
+ * A component for view, search and select a list of items
+ * TODO select will be introduced with story https://www.pivotaltracker.com/story/show/167102335
+ */
+export class ChooserListContainer<T> extends React.PureComponent<
+  Props<T>,
+  State
+> {
+  constructor(props: Props<T>) {
+    super(props);
+    this.state = {
+      searchText: this.props.isSearchEnabled ? some("") : none,
```

Comment:
Could you leave a comment describing these variables?

Reply:
Done in commit 5ba4dc4